src/router/mod.rs

```diff
@@ -106,7 +106,7 @@ impl Router {
     }
 
     fn handle_routing_update(&mut self, logger: &Logger, routing_response: &RoutingResponse) {
-        if routing_response.height <= self.routing_height {
+        if routing_response.height < self.routing_height {
```

Comment:
mostly because you've already processed the state at that block. Are you seeing issues where it's returning the same block again? Seems odd